pkg/controller/externalnode/validate.go

```diff
@@ -0,0 +1,82 @@
+// Copyright 2022 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package externalnode
+
+import (
+	"encoding/json"
+	"fmt"
+
+	admv1 "k8s.io/api/admission/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/klog/v2"
+
+	crdv1alpha1 "antrea.io/antrea/pkg/apis/crd/v1alpha1"
+)
+
+func (c *ExternalNodeController) ValidateExternalNode(review *admv1.AdmissionReview) *admv1.AdmissionResponse {
+	var result *metav1.Status
+	var msg string
+	allowed := true
+
+	klog.V(2).Info("Validating ExternalNode", "request", review.Request)
+	var newObj crdv1alpha1.ExternalNode
+	if review.Request.Object.Raw != nil {
+		if err := json.Unmarshal(review.Request.Object.Raw, &newObj); err != nil {
+			klog.ErrorS(err, "Error de-serializing current ExternalNode", "name", newObj.Name)
+			return newAdmissionResponseForErr(err)
+		}
+	}
+
+	shouldAllow := func(externalNode *crdv1alpha1.ExternalNode) (bool, string) {
+		if len(externalNode.Spec.Interfaces) == 0 {
+			return false, fmt.Sprintf("ExternalNode %s does not specify interface", externalNode.Name)
+		}
+		if len(externalNode.Spec.Interfaces[0].IPs) == 0 {
+			return false, fmt.Sprintf("ExternalNode %s Interface[0] does not specify IP", externalNode.Name)
+		}
```

Comment:
I think the validations can be implemented by openAPIV3Schema: minItems and maxItems attributes.

Reply:
Thanks a lot for the information! I will investigate this part.

Reply:
Since we don't have the maximum requirement, I marked the corresponding fields as required. Please help review the latest version, thanks!